pumps/mongo_aggregate.go

```diff
@@ -373,6 +307,78 @@ func (m *MongoAggregatePump) HandleWriteErr(err error) error {
 	return err
 }
 
+func (m *MongoAggregatePump) DoAggregatedWriting(ctx context.Context, orgID string, filteredData analytics.AnalyticsRecordAggregate) error {
+	collectionName, collErr := m.GetCollectionName(orgID)
+	if collErr != nil {
+		m.log.Info("No OrgID for AnalyticsRecord, skipping")
+		return nil
+	}
+
+	thisSession := m.dbSession.Copy()
+	defer thisSession.Close()
+
+	analyticsCollection := thisSession.DB("").C(collectionName)
+	indexCreateErr := m.ensureIndexes(analyticsCollection)
+
+	if indexCreateErr != nil {
+		m.log.Error(indexCreateErr)
```

Comment:
should we return err?

Reply:
@sredxny before it didn't cut the writing execution if the was an error creating indexes. I don't want to change that logic, IMO we should keep trying to write the records despite the index

Reply:
ok, got it